src/hisp/plamsa_data_handling/main.py

```diff
@@ -197,21 +208,37 @@ def get_heat(self, pulse: Pulse, bin: SubBin | DivBin, t_rel: float):
             raise ValueError(f"Invalid pulse type {pulse.pulse_type}")
 
         if pulse.pulse_type == "FP":
-            heat_total = data[:, -2][bin_index]
-            heat_ion = data[:, -1][bin_index]
+            heat_total = data["heat_total"][bin_index]
+            heat_ion = data["heat_ion"][bin_index]
             if isinstance(bin, SubBin):
                 heat_val = heat_total - heat_ion * (1 - bin.wetted_frac)
             else:
                 heat_val = heat_total
         elif pulse.pulse_type == "RISP":
             if isinstance(bin, SubBin):
-                heat_total = data[-2]
-                heat_ion = data[-1]
+                heat_total = data["heat_total"]
+                heat_ion = data["heat_ion"]
                 heat_val = heat_total - heat_ion * (1 - bin.wetted_frac)
             else:
-                heat_val = data[-1]
+                heat_val = data["heat_total"]
+
+            # if heat_val is an empty pandas Series set it at 0.0 (no heat)
+            # otherwise take the single value from the series
+            if isinstance(heat_val, pd.Series):
+                if heat_val.values.size == 0:
+                    heat_val = 0.0
+                else:
+                    assert (
+                        heat_val.values.size == 1
+                    ), f"heat_val should be a single value, values: {heat_val.values}"
+                    heat_val = heat_val.values[0]
```

Comment:
I noticed that there is a difference across pulse types here. For FP, `data` contains all bins (hence we need to use `[bin_index]`. This is not the case for RISP.

We can maybe simplify this function greatly if we modify the `RISP_data()` method to return the whole dataframe (for all bins in the file) and not slice it. At least this would make this more understandable